src/main/java/de/dennisguse/opentracks/adapters/IntervalStatisticsAdapter.java

```diff
@@ -17,60 +16,78 @@
 
-public class IntervalStatisticsAdapter extends ArrayAdapter<IntervalStatistics.Interval> {
+public class IntervalStatisticsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
 
+    private List<IntervalStatistics.Interval> intervalList;
+    private final Context context;
     private final StackMode stackMode;
-    private final boolean metricUnits;
-    private final String category;
+    private boolean metricUnits;
+    private String category;
 
-    public IntervalStatisticsAdapter(Context context, List<IntervalStatistics.Interval> intervalList, String category, StackMode stackMode) {
-        super(context, R.layout.interval_stats_list_item, intervalList);
+    public IntervalStatisticsAdapter(Context context, String category, StackMode stackMode) {
         metricUnits = PreferencesUtils.isMetricUnits(PreferencesUtils.getSharedPreferences(context), context);
+        this.context = context;
         this.category = category;
         this.stackMode = stackMode;
     }
 
-    //TODO Check preference handling! Should not be accessed in getView()
     @NonNull
     @Override
-    public View getView(int position, @Nullable View intervalView, @NonNull ViewGroup parent) {
-        int actualPosition = stackMode == StackMode.STACK_FROM_TOP ? position : getCount() - 1 - position;
-        IntervalStatistics.Interval interval = getItem(actualPosition);
-        ViewHolder viewHolder;
-
-        if (intervalView == null) {
-            viewHolder = new ViewHolder();
-
-            intervalView = LayoutInflater.from(getContext()).inflate(R.layout.interval_stats_list_item, parent, false);
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.interval_stats_list_item, parent, false);
+        return new IntervalStatisticsAdapter.ViewHolder(view);
+    }
 
-            viewHolder.distance = intervalView.findViewById(R.id.interval_item_distance);
-            viewHolder.rate = intervalView.findViewById(R.id.interval_item_rate);
-            viewHolder.gain = intervalView.findViewById(R.id.interval_item_gain);
-            viewHolder.loss = intervalView.findViewById(R.id.interval_item_loss);
+    //TODO Check preference handling! Should not be accessed in getView()
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+        int actualPosition = stackMode == StackMode.STACK_FROM_TOP ? position : getItemCount() - 1 - position;
+        IntervalStatisticsAdapter.ViewHolder viewHolder = (IntervalStatisticsAdapter.ViewHolder) holder;
+        IntervalStatistics.Interval interval = intervalList.get(actualPosition);
+        viewHolder.itemView.setTag(actualPosition);
 
-            intervalView.setTag(viewHolder);
+        float sumDistance_m;
+        if (actualPosition + 1 == getItemCount() && actualPosition > 0) {
```

Comment:
can we use a variable for this aka `isLast = actualPosition == getItemCount() - 1`

Reply:
and also nextPostion = actualPosition + 1